packages/react/src/useResource/useResource.ts

```diff
@@ -23,6 +23,10 @@ export function useResource<T extends Resource>(value: Reference<T> | T | undefi
   const medplum = useMedplum();
   const [resource, setResource] = useState<T | undefined>(getInitialResource(medplum, value));
 
+  useEffect(() => {
```

Comment:
IIRC, the original implementation used this exact pattern, which resulted in unnecessary rerenders.  Removing `useEffect` may have been too far.

I agree that stale data is a worse problem than unnecessary renders, so I'm ok with this.

I wonder if there's a check we could add inside the effect to get the best of both worlds.

For example:

```ts
const initialResource = getInitialResource(medplum, value);
if (initialResource?.meta?.version !== resource?.meta?.version) {
  setResource(initialResource);
}

Reply:
I think that might break in cases where the version might the same, but the full object is different. 

One case is modifying a resource on the client-side in editing workflows. I just added a test for this:

Another case would be when first `value` is a partial resource from GraphQL, and then later is a full resource from `readResource()`.